bpf/lib/nodeport.h

```diff
@@ -1167,29 +1182,27 @@ static __always_inline bool snat_v4_needed(struct __ctx_buff *ctx, __be32 *addr,
 			 */
 			einfo = lookup_ip4_egress_endpoint(ip4->saddr, ip4->daddr);
 			if (einfo) {
-				*addr = einfo->egress_ip;
-				*from_endpoint = true;
-				return true;
+				/* If the packet is a reply it means that the
+				 * connection did not originate from a pod,
+				 * and so we should skip SNAT (similar to how we
+				 * skipped forwarding the packet over the tunnel
+				 * to the egress gateway in bpf_lxc)
+				 */
+				if (!ct_is_reply) {
```

Comment:
Shouldn't we check *before* the egress map lookup, to avoid an unnecessary lookup?